### PR TITLE
Admin views

### PR DESCRIPTION
**Pull request by tzelic**

### CODE REVIEW

**Comment by tzelic** [comment]
Made all views for admin

**Comment by b0rza** [comment]
Revert back to using default scope before merging to master!

File: server/models/totalOodlet.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const Oodler = require('./oodler');
+
+const schema = {
+  id: type.string(),
+  createdAt: type.date().default(() => new Date()),
+  updatedAt: type.date().default(() => new Date()),
+  orderedAt: type.date(),
+  oodler: type.object().schema(Oodler.schema),
+  quantifiedThingies: [type.object().schema(object.merge(Thingy.schema, { qty: type.number() }))],
+  oodletIds: type.array(type.string())
```
Comment (b0rza) [inline server/models/totalOodlet.js]: Use the same array declaration method for `quantifiedThingies` and `oodletIds`.

File: server/routes/oodler.js [hunk 2]
```diff
@@ -64,7 +64,7 @@ function update(request, reply) {
           lastName: request.payload.lastName,
           email: request.payload.email,
           office: request.payload.office,
-          scope: request.payload.office
+          scope: request.payload.scope
```
Comment (b0rza) [inline server/routes/oodler.js]: Do it on line 39 as well ;)

File: server/routes/oodlet.js [hunk 3]
```diff
@@ -27,11 +28,25 @@ function get(request, reply) {
     });
 }
 
-function create(request, reply) {
+function active(request, reply) {
+  oodletUtil.findActive(request.query.office)
+    .then((activeOodlets) => {
```
Comment (b0rza) [inline server/routes/oodlet.js]: Parenthesis not needed.

File: server/routes/totalOodlet.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+const oodletUtil = require('../util/oodlet');
+const totalOodletUtil = require('../util/totalOodlet');
+
+function list(request, reply) {
+  let fromDate = (() => { return request.query.fromDate ? moment(request.query.fromDate).toDate() : moment().subtract(3, 'months').toDate(); })();
+  let toDate = (() => { return request.query.fromDate ? moment(request.query.toDate).add(1, 'days').toDate() : moment().toDate(); })();
+  
+  return TotalOodlet
+    .between(fromDate, toDate, { index : 'orderedAt' })
+    .filter(function (row) {
```
Comment (b0rza) [inline server/routes/totalOodlet.js]: ANONYMOUS

File: server/util/oodlet.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+function create(oodler, dueDate) {
+  return Oodlet({
+      oodler: oodler,
+      dueDate: dueDate
+    })
+    .save();
+}
+
+function nextDueDate() {
```
Comment (b0rza) [inline server/util/oodlet.js]: :+1:

File: client/components/OodlerEditTile.vue [hunk 6]
```diff
@@ -0,0 +1,176 @@
+<template>
+  <div class="oodler-edit-tile">
+    <div class="thumbnail">
```
Comment (tzelic) [inline client/components/OodlerEditTile.vue]: I need some kind of container, but I can use `well` to make it more logical. Effect is the same.

File: client/components/OodlerEditTile.vue [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+  
+  export default {
+    data() {
+      return {
```
Comment (tzelic) [inline client/components/OodlerEditTile.vue]: For `editOodler` I agree: `editedOodler` would be better (I already have var `oodler` as a prop).
But for `editing` object it's not action, it's a state (true or false i.e. is firstName's state editing or not?) Why is that wrong approach?

**Comment by tzelic** [comment]
I agree with nesting views and components as admin, user or shared, but was waiting for you to decide that.
About devs accs it is done in #46

File: client/views/OodlersView.vue [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  <div id="oodlersView">
+    <div class="page-header">
+      <h1 class="text-info">Oodlers</h1>
+    </div>
+    <div class="row">
+      <div class="filtered-oodlers col-md-12">
+        <div class="row">
+          <search-bar class="col-md-12" subject="oodler by first name, last name or office" @searchBarUpdate="searchBarUpdate"></search-bar>
+        </div>
+        <ul class="row">
```
Comment (tzelic) [inline client/views/OodlersView.vue]: In this case I am droping list altogether :+1:

File: client/components/HistoryTotalOodlet.vue [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+      <div class="panel-heading">
+        <strong>Ordered on: </strong>
+        <span>{{ orderedOn }}</span>
+      </div>
+      <div class="panel-body">
+        <table class="table table-striped table-hover">
+          <tbody>
+          <tr v-for="quantifiedThingy in totalOodlet.quantifiedThingies">
+            <td>{{ quantifiedThingy.name }}</td>
+            <td class="col-md-2 right">{{ quantifiedThingy.qty }}</td>
```
Comment (b0rza) [inline client/components/HistoryTotalOodlet.vue]: :+1: 
OK, let's leave it for a future style update/refactor.

File: client/components/OodlerEditTile.vue (hunk 7 above)
Comment (b0rza) [inline client/components/OodlerEditTile.vue]: Appreciate the push back :+1:

File: server/routes/totalOodlet.js [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+const oodletUtil = require('../util/oodlet');
+const totalOodletUtil = require('../util/totalOodlet');
+
+function list(request, reply) {
+  let fromDate = (() => { return request.query.fromDate ? moment(request.query.fromDate).toDate() : moment().subtract(3, 'months').toDate(); })();
+  let toDate = (() => { return request.query.fromDate ? moment(request.query.toDate).add(1, 'days').toDate() : moment().toDate(); })();
+  
+  return TotalOodlet
+    .between(fromDate, toDate, { index : 'orderedAt' })
+    .filter(row => {
```
Comment (b0rza) [inline server/routes/totalOodlet.js]: Single line, no brackets needed or `return` needed

**Pull request merged**